### PR TITLE
added EP to ack notification

### LINKED ISSUE #1296: Implement "seen" for notifications
Usually messaging systems have an ack functionality for messages, try to see if we have it with google.

if not, we might need to implement a seen field in notifications

### PR DESCRIPTION
**Pull request by noamso1**

### CODE REVIEW

**Comment by noamso1** [comment]
Done. view example on POSTMAN: Notification / Seen

File: functions/src/notification/business/markSeen.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const admin = require('firebase-admin');
+const db = admin.firestore();
+
+import { INotificationModel } from '../../shared/interfaces/INotificationModel';
+import { Nullable } from '../../shared/types/Nullable';
+import { CommonError } from '../../util/errors';
+const COLLECTION_NAME = 'notification';
+
+export const markSeen = async (id: string, userId: string) => {
+  let notification : INotificationModel = (await db.collection(COLLECTION_NAME).doc(id).get()).data() as Nullable<INotificationModel>;
```
Comment (ariel-dao) [inline functions/src/notification/business/markSeen.ts]: why is this Nullable?

Comment (ariel-dao) [inline functions/src/notification/business/markSeen.ts]: change let to const

File: functions/src/notification/controllers/markSeen.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { responseExecutor } from '../../util/responseExecutor';
+import { markSeen as markSeenDb  } from '../business/markSeen';
+
+export const markSeen = async (req, res, next) => {
+    await responseExecutor(
+        async () => {
+          return await markSeenDb(req.body.id, req.user.uid);
```
Comment (ariel-dao) [inline functions/src/notification/controllers/markSeen.ts]: remove await

File: functions/src/notification/business/markSeen.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  if ( !notification.userFilter.includes( userId )) {
```
Comment (ariel-dao) [inline functions/src/notification/business/markSeen.ts]: validation should be made on the controller level

File: functions/src/notification/business/markSeen.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+const admin = require('firebase-admin');
+const db = admin.firestore();
+
+import { INotificationModel } from '../../shared/interfaces/INotificationModel';
+import { Nullable } from '../../shared/types/Nullable';
+import { CommonError } from '../../util/errors';
+const COLLECTION_NAME = 'notification';
```
Comment (ariel-dao) [inline functions/src/notification/business/markSeen.ts]: use enum

File: functions/src/notification/business/markSeen.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+export const markSeen = async (id: string, userId: string) => {
+  let notification : INotificationModel = (await db.collection(COLLECTION_NAME).doc(id).get()).data() as Nullable<INotificationModel>;
+
+  if ( !notification.userFilter.includes( userId )) {
+    throw new CommonError('Notification does not belong to you', { });
+  }
+
+  notification.seen = true
+  await db.collection(COLLECTION_NAME).doc(id).update(notification)
```
Comment (ariel-dao) [inline functions/src/notification/business/markSeen.ts]: enum

File: functions/src/notification/business/markSeen.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import { INotificationModel } from '../../shared/interfaces/INotificationModel';
+import { Nullable } from '../../shared/types/Nullable';
+import { CommonError } from '../../util/errors';
+const COLLECTION_NAME = 'notification';
+
+export const markSeen = async (id: string, userId: string) => {
+  let notification : INotificationModel = (await db.collection(COLLECTION_NAME).doc(id).get()).data() as Nullable<INotificationModel>;
+
+  if ( !notification.userFilter.includes( userId )) {
+    throw new CommonError('Notification does not belong to you', { });
```
Comment (ariel-dao) [inline functions/src/notification/business/markSeen.ts]: object should consist details and not be empty

File: functions/src/notification/business/markSeen.ts (hunk 1 above)
Comment (noamso1) [reply functions/src/notification/business/markSeen.ts]: Done - removed Nullable

Comment (noamso1) [reply functions/src/notification/business/markSeen.ts]: done

File: functions/src/notification/controllers/markSeen.ts (hunk 2 above)
Comment (noamso1) [reply functions/src/notification/controllers/markSeen.ts]: Done

File: functions/src/notification/business/markSeen.ts (hunk 4 above)
Comment (noamso1) [reply functions/src/notification/business/markSeen.ts]: Done

File: functions/src/notification/business/markSeen.ts (hunk 6 above)
Comment (noamso1) [reply functions/src/notification/business/markSeen.ts]: Done

File: functions/src/notification/business/markSeen.ts (hunk 3 above)
Comment (noamso1) [reply functions/src/notification/business/markSeen.ts]: Done

File: functions/src/notification/business/markSeen.ts (hunk 5 above)
Comment (noamso1) [reply functions/src/notification/business/markSeen.ts]: Done

**Review by ariel-dao — APPROVED** [review]

**Pull request merged**